### PR TITLE
Allow aeson 2.0

### PR DESCRIPTION
**Pull request by jvanbruegge**
Only needed changes to the tests, the main library is untouched.

Also updates the stack.yaml to use GHC 9.0.2

### CODE REVIEW

File: haxl.cabal [hunk 1]
```diff
@@ -43,7 +43,7 @@ extra-source-files:
 library
 
   build-depends:
-    aeson >= 0.6 && < 1.6,
+    aeson >= 0.6 && < 2.1,
```
[the remark this answers is not in the document]
Comment (jvanbruegge) [reply haxl.cabal:46]: The main library does not use `KeyMap` or `HashMap` directly, so it works for 1.5 and 2.0.

Only the tests use the keymap directly. I can of course at `CPP` to the tests, but with 2.0 allowed, cabal will always use aeson 2.0 for the tests.

TL;DR library is compatible with both aeson-1.5 and aeson-2.0 after this PR

[the remark this answers is not in the document]
Comment (watashi) [reply haxl.cabal:46]: But this won't work if you have stack or other build system with older version of aeson specified.
Particularly, Facebook's Haskell env hasn't been upgrade to aeson-2.0 so the tests won't build.
Can you guard the change with `CPP` or rewrite it in an compatible way?

[the remark this answers is not in the document]
Comment (pepeiborra) [reply haxl.cabal:46]: Yeah, I'm afraid that we need the tests to work with the version of aeson in LTS 16.

**Comment by jvanbruegge** [comment]
I've added CPP to the tests now. I verified that this still builds with stack LTS 19 as well as 18 (so pre-aeson-2.0)

**Pull request closed without merging**